cloud/ec2_client.go

```diff
@@ -46,6 +47,12 @@ type AWSClient interface {
 	// DescribeSpotInstanceRequests is a wrapper for ec2.DescribeSpotInstanceRequests.
 	DescribeSpotInstanceRequests(context.Context, *ec2.DescribeSpotInstanceRequestsInput) (*ec2.DescribeSpotInstanceRequestsOutput, error)
 
+	// DescribeSpotRequestsAndSave is a wrapper for DescribeSpotInstanceRequests that also saves instance IDs to the db
+	DescribeSpotRequestsAndSave(context.Context, []*host.Host) (*ec2.DescribeSpotInstanceRequestsOutput, error)
```

Comment:
With the addition of this, we should probably not have the method on line 48 and use this everywhere?

Reply:
done